addons/blocklydevtools/style.css

```diff
@@ -0,0 +1,26 @@
+/* Hide the ✏️ button on blocks that aren't top level */
+g.blocklyDraggable[data-id]:not(.blocklyInsertionMarker):not(g.blocklyBlockCanvas>g.blocklyDraggable[data-id]:not(.blocklyInsertionMarker)) [data-is-blocklydev-btn] {
+    display: none !important;
+}
+
+/* Hide editor buttons on blocks that aren't top level */
+g.blocklyDraggable[data-id]:not(.blocklyInsertionMarker):not(g.blocklyBlockCanvas>g.blocklyDraggable[data-id]:not(.blocklyInsertionMarker)) [data-is-blocklydev-editor-btn] {
+    display: none !important;
+}
+
+/* Hide the XML editor on blocks that aren't top level */
+g.blocklyDraggable[data-id]:not(.blocklyInsertionMarker):not(g.blocklyBlockCanvas>g.blocklyDraggable[data-id]:not(.blocklyInsertionMarker)) [data-isblocklydeveditor] {
+    display: none !important;
+}
```

Comment:
Could these be given classes that start with `sa-`? These can probably be combined into 1 rule. Also, have you checked if `!important` is actually necessary?

Reply:
It is, because the buttons use inline styles, which by default are considered more important than seperate style tags

Reply:
I can't rename the blockly internal classes, but I will store the button and editor css as a class soon.

Reply:
Done merging css rules.